graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java

```diff
@@ -279,14 +279,14 @@ private String ensureCollector(String collectorName,
                                    String defaultTemplate) {
         Collector collector = null;
         try {
-            collector = collectorService.findByName(collectorName);
+            collector = collectorService.findByNameAndOs(collectorName, nodeOperatingSystem);
             if (collector == null) {
-                final String msg = "Couldn't find collector '" + collectorName + "' fixing it.";
-                LOG.error(msg);
+                final String msg = "Couldn't find collector '{} on {}' fixing it.";
+                LOG.error(msg, collectorName, nodeOperatingSystem);
                 throw new IllegalArgumentException(msg);
```

Comment:
The `msg` here is just a log template and doesn't have the `collectorName` and `nodeOperatingSystem` in the exception message.

Reply:
hmm, it is ignored anyway :-)
but yeah, better fix it